plugins/woocommerce/src/Admin/API/Reports/TimeInterval.php

```diff
@@ -623,4 +623,80 @@ public static function rest_validate_between_date_arg( $value, $request, $param
 
 		return true;
 	}
+
+	/**
+	 * Get dates from a timeframe string.
+	 *
+	 * @param int           $timeframe Timeframe to use.  One of: last_week|last_month|last_quarter|last_6_months|last_year.
+	 * @param DateTime|null $current_date DateTime of current date to compare.
+	 * @return array
+	 */
+	public static function get_timeframe_dates( $timeframe, $current_date = null ) {
```

Comment:
Nice! I am rather surprised we didn't have this already, but I guess we handle this part on the front-end in Analytics.

Reply:
I was equally surprised 🤷